File: WWF/Hands on labs 3.0/Lab01/Completed/Exercise 4/ExpenseWorkflows/Workflow1.xoml.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace ExpenseWorkflows
{
	public partial class Workflow1 : SequentialWorkflowActivity
	{
        public string ReportEmployeeId = default(System.String);
        public string ManagerEmployeeId = default(System.String);
        public ExpenseLocalServices.ExpenseReportReviewedEventArgs reviewArgs =
            default(ExpenseLocalServices.ExpenseReportReviewedEventArgs);


        public int amount = default(System.Int32);
        public ExpenseLocalServices.ExpenseReportSubmittedEventArgs reportArgs = default(ExpenseLocalServices.ExpenseReportSubmittedEventArgs);
       
        private void ReportSubmitted_Invoked(object sender, ExternalDataEventArgs e)
        {
            Console.WriteLine("ReportSubmitted_Invoked");

            this.amount = this.reportArgs.Report.Amount;

            this.ReportEmployeeId = this.reportArgs.Report.EmployeeId;
        }

        private void IfReportApproved_Condition(object sender, ConditionalEventArgs e)
        {
            e.Result = this.reviewArgs.Review.Approved;
        }
	}
}

File: WWF/Hands on labs 3.0/VB/Lab01/Completed/Exercise 4/ExpenseWorkflows/ExpenseWorkflows/Workflow1.xoml.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

'NOTE: When changing the namespace; please update XmlnsDefinitionAttribute in AssemblyInfo.vb
Public Class Workflow1
    Inherits SequentialWorkflowActivity


    Public reportArgs As ExpenseLocalServices.ExpenseReportSubmittedEventArgs
    Public Amount As Integer

    Public ReportEmployeeId As String
    Public ManagerEmployeeId As String
    Public reviewArgs As ExpenseLocalServices.ExpenseReportReviewedEventArgs

    Private Sub ReportSubmitted_Invoked(ByVal sender As System.Object, _
                            ByVal e As System.Workflow.Activities.ExternalDataEventArgs)
        Console.WriteLine("ReportSubmitted_Invoked")
        Me.Amount = Me.reportArgs.Report.Amount
        Me.ReportEmployeeId = Me.reportArgs.Report.EmployeeId
    End Sub

    Private Sub IfReportApproved_Condition(ByVal sender As System.Object, _
                        ByVal e As System.Workflow.Activities.ConditionalEventArgs)
        e.Result = Me.reviewArgs.Review.Approved
    End Sub
End Class
